### PR TITLE
Implemented query builder

### PR DESCRIPTION
**Pull request by Waguramu**
(ﾉ◕ヮ◕)ﾉ*:･ﾟ✧ **Implemented the initial QueryBuilder Functionality**

(^=◕ᴥ◕=^)ﾉ˚(っ˘ω˘ς )(^=◕ᴥ◕=^)

### CODE REVIEW

File: src/util/query_builder.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    def set_values(self, properties: Dict[str, str], letter: str) -> QueryBuilder:
+        """
+        Create Cypher SET query to set the value to the node in the query
+        :param properties: properties that have to be initialised
+        :param letter: a variable identifying the node in the Cypher query
+        :return: this QueryBuilder
+        """
+        for key, value in properties.items():
+            self.add("SET " + letter + "." + key + "='" + value + "'")
```
Comment (josephbirkner) [inline src/util/query_builder.py]:
```suggestion
            self.add(f"SET {letter}.{key} = '{value}'")
```

Comment (josephbirkner) [reply src/util/query_builder.py]: Do we need to care about escaping the value?

Comment (Waguramu) [reply src/util/query_builder.py]: Well, this 
`self.add(f"{{{', '.join(param_list)}}}")`
looks fugly. Shall we proceed? Or use 
`'%λ' % lambda`

File: src/util/query_builder.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        self.query.join("".join(parts))
+        return self
+
+    def add(self, text: str) -> QueryBuilder:
+        """
+        Add a single token/call into the Cypher query
+        :param text: string containing a single part of the query
+        :return: this QueryBuilder
+        """
+        return self.append([" ", text, " "])
```
[the remark this answers is not in the document]
Comment (Waguramu) [reply src/util/query_builder.py]: Not of type List[str]

File: src/util/query_builder.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        """
+        return self.add(text + ''.join([str(x) for x in args]))
+
+    def add_parameters(self, properties: Dict[str, str]) -> QueryBuilder:
+        """
+        Add the node properties to the Cypher query
+        :param properties: properties of the node
+        :return: this QueryBuilder
+        """
+        if properties is None:
```
[the remark this answers is not in the document]
Comment (Waguramu) [reply src/util/query_builder.py]: Removed

File: src/util/query_builder.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        return self.append([" ", text, " "])
+
+    def add_arguments(self, text: str, args: List[object]) -> QueryBuilder:
+        """
+        Add arguments with a token/call to the Cypher query
+        :param text: token/call defining the list of objects
+        :param args: list of objects to be added to the token
+        :return: this QueryBuilder
+        """
+        return self.add(text + ''.join([str(x) for x in args]))
```
[the remark this answers is not in the document]
Comment (Waguramu) [reply src/util/query_builder.py]: Was misdone
Changed to reflect the original
May be redundant

File: src/util/query_builder.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+    def add(self, text: str) -> QueryBuilder:
+        """
+        Add a single token/call into the Cypher query
+        :param text: string containing a single part of the query
+        :return: this QueryBuilder
+        """
+        return self.append([" ", text, " "])
+
+    def add_arguments(self, text: str, args: List[object]) -> QueryBuilder:
```
[the remark this answers is not in the document]
Comment (Waguramu) [reply src/util/query_builder.py]: Was misdone
Changed to reflect the original
May be redundant

**Pull request merged**